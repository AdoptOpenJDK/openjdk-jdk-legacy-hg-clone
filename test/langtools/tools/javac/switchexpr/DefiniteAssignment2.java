/**
 * @test /nodynamiccopyright/
 * @bug 8214031
 * @summary Verify that definite assignment when true works (illegal code)
 * @compile/fail/ref=DefiniteAssignment2.out --enable-preview --source ${jdk.version} -XDrawDiagnostics DefiniteAssignment2.java
 */
public class DefiniteAssignment2 {

    public static void main(String[] args) {
        int a = 0;
        boolean b = true;
        boolean t;

        {
            int x;

            t = (b && switch(a) {
                case 0: break (x = 1) == 1 || true;
                default: break false;
            }) || x == 1;
        }

        {
            int x;

            t = (switch(a) {
                case 0: break (x = 1) == 1;
                default: break false;
            }) || x == 1;
        }

        {
            int x;

            t = (switch(a) {
                case 0: x = 1; break true;
                case 1: break (x = 1) == 1;
                default: break false;
            }) || x == 1;
        }

        {
            int x;

            t = (switch(a) {
                case 0: break true;
                case 1: break (x = 1) == 1;
                default: break false;
            }) && x == 1;
        }

        {
            int x;

            t = (switch(a) {
                case 0: break false;
                case 1: break isTrue() || (x = 1) == 1;
                default: break false;
            }) && x == 1;
        }

        {
            int x;

            t = (switch(a) {
                case 0: break false;
                case 1: break isTrue() ? true : (x = 1) == 1;
                default: break false;
            }) && x == 1;
        }

        {
            final int x;

            t = (switch(a) {
                case 0: break false;
                case 1: break isTrue() ? true : (x = 1) == 1;
                default: break false;
            }) && (x = 1) == 1;
        }
    }

    private static boolean isTrue() {
        return true;
    }

}
