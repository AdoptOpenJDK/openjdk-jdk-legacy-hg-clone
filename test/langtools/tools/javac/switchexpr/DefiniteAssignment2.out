DefiniteAssignment2.java:20:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:29:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:39:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:49:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:59:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:69:19: compiler.err.var.might.not.have.been.initialized: x
DefiniteAssignment2.java:79:20: compiler.err.var.might.already.be.assigned: x
- compiler.note.preview.filename: DefiniteAssignment2.java
- compiler.note.preview.recompile
7 errors
